lib/verify-vsce-auth.js

```diff
@@ -5,16 +5,29 @@ const execa = require('execa');
 
 module.exports = async (logger, cwd) => {
   logger.log('Verifying authentication for vsce');
+  logger.log(`VSCE_AZURE_CREDENTIALS: ${process.env.VSCE_AZURE_CREDENTIALS}`);
 
-  if (!process.env.VSCE_PAT) {
+  if (!process.env.VSCE_PAT && !process.env.VSCE_AZURE_CREDENTIALS) {
     throw new SemanticReleaseError(
-      'Empty vsce personal access token (`VSCE_PAT` environment variable) specified.',
-      'EEMPTYVSCEPAT',
+      'Neither vsce personal access token (`VSCE_PAT` environment variable) or Azure Credentials flag (`VSCE_AZURE_CREDENTIALS` environment variable) specified.',
+      'EVSCEAUTHNOTPROVIDED',
     );
   }
```

Comment:
You'll have to change this a little. This actually checks whether the variable is present but empty, like: VSCE_PAT="".

Maybe you should invent two cases and two error codes? One to check for empty vsce pat and another to check for empty VSCE_AZURE_CREDENTIALS.

if ('VSCE_PAT' in process.env && !process.env.VSCE_PAT) {
  // old condition
}

// something similar for AZURE_CREDENTIALS